File: 02-task-command-injection/vuln_code.py
#!/usr/bin/env python3

"""
Exercise 02 - Command Injection
WORKSHOP: Secure Programming

Audit: Read this code carefully. Can you spot the security flaw?
"""

import os
import subprocess


def ping_host(host):
    """Ping a host to check if it is alive."""
    result = subprocess.run(
        ["ping", "-c", "1", host],
        capture_output=True,
        text=True,
    )
    if result.returncode == 0:
        print("[+] Host Responds.")
    else:
        print("[-] Host did not respond.")

if __name__ == "__main__":
    host = input("Enter IP address to ping: ")
    ping_host(host)

File: 05-tutorial-buffer-overflow/vuln_code.c
#include <stdio.h>
#include <string.h>

/*
 * Tutorial 05 - Buffer Overflow
 * WORKSHOP: Secure Programming
 *
 * Audit: Read this code carefully. Can you spot the security flaw?
 */

void greet_user(void)
{
	char buffer[128];   /* Only 128 bytes allocated */

	printf("What is your name? ");
	fflush(stdout);

	fgets(buffer, 128, stdin);      /* UNSAFE: reads past buffer size - no bounds check! */

	printf("Hello, %s!\n", buffer);
}

int main(void)
{
	greet_user();

	return 0;
}
